Upgrade gspell to 1.14.0-1

kevin-b-lockwood opened the pull request:
<!--
COMMENT BLOCKS WILL NOT BE INCLUDED IN THE PR.
Feel free to delete sections of the template which do not apply to your PR, or add additional details
-->

###### Merge Checklist  <!-- REQUIRED -->
<!-- You can set them now ([x]) or set them later using the Github UI -->
**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary <!-- REQUIRED -->
<!-- Quick explanation of the changes. -->
Upgrades gspell to current upstream

###### Build/Dependency Information <!-- REQUIRED -->
Please build the following packages with gspell
iso-codes

###### Change Log  <!-- REQUIRED -->
<!-- Detail the changes made here. -->
<!-- Please list any packages which will be affected by this change, if applicable. -->
<!-- Please list any CVES fixed by this change, if applicable. -->
- Switch from `configure` and `make` to `meson`
- License Verified
- Change License to LGPLv2.1+

###### Does this affect the toolchain?  <!-- REQUIRED -->
<!-- Any packages which are included in the toolchain should be carefully considered. Make sure the toolchain builds with these changes if so. -->
<!-- Update: manifests/package/toolchain_*.txt, pkggen_core_*.txt, update_manifests.sh -->
<!-- To validate: make clean; make workplan REBUILD_TOOLCHAIN=y DISABLE_UPSTREAM_REPOS=y CONFIG_FILE="" ... -->
**NO**

###### Test Methodology
<!-- How was this test validated? i.e. local build, pipeline build etc. -->
- Local build

kevin-b-lockwood committed:
Upgrade gspell to 1.14.0-1

0xba1a commented on SPECS-EXTENDED/gspell/gspell.spec:
```diff
@@ -4,16 +4,17 @@ Distribution:   Azure Linux
-Source0:        https://download.gnome.org/sources/%{name}/1.8/%{name}-%{version}.tar.xz
+License:        LGPLv2.1+
+URL:            https://gitlab.gnome.org/GNOME/gspell
+Source0:	https://gitlab.gnome.org/GNOME/gspell/-/archive/%{version}/%{name}-%{version}.tar.gz
```
Please replace tab with spaces

kevin-b-lockwood replied:
My editor was not configured to show non-printing characters, I'll fix that now.

0xba1a commented on SPECS-EXTENDED/gspell/gspell.spec:
```diff
@@ -4,16 +4,17 @@ Distribution:   Azure Linux
-URL:            https://wiki.gnome.org/Projects/gspell
-Source0:        https://download.gnome.org/sources/%{name}/1.8/%{name}-%{version}.tar.xz
+License:        LGPLv2.1+
+URL:            https://gitlab.gnome.org/GNOME/gspell
```
Please try to use `gnome.org` for fetching source for security purposes.

kevin-b-lockwood replied:
I was simply following the link that was provided on the gnome.org website:
![image](https://github.com/user-attachments/assets/92c9e4c1-aea5-4d0d-8d79-7af9a3e0e4bc)
I am happy to change it back if that is preferred, though.

0xba1a commented on SPECS-EXTENDED/gspell/gspell.spec:
```diff
@@ -67,9 +68,9 @@ find $RPM_BUILD_ROOT -name '*.la' -delete
+%license LICENSES/*
 %{_libdir}/girepository-1.0/
-%{_libdir}/libgspell-1.so.2*
+%{_libdir}/libgspell-1.so.3*
```
This looks like a breaking change. Did you check other depending on packages whether they're building and running properly?

kevin-b-lockwood replied:
I have grepped the SPECS (`rg Requires:.*gspell`, `rg gspell`) and have not come across a package that depends on `gspell`. It seems nothing depends on it. Please advise if I am mistaken.

0xba1a commented on SPECS-EXTENDED/gspell/gspell.spec:
```diff
@@ -78,12 +79,19 @@ find $RPM_BUILD_ROOT -name '*.la' -delete
 %{_libdir}/pkgconfig/gspell-1.pc
 %{_datadir}/gir-1.0/
 %{_datadir}/vala/
+%{_datadir}/installed-tests/%{name}-1/
```
We don't need test files as they'll unnecessarily bloat the rpm size.

kevin-b-lockwood replied:
I will exclude those files.

kevin-b-lockwood committed:
(squash) Fix build with tests

I neglected to check earlier if this built with tests it seems, and as a
result I have to fix it now.
